src/pemfile.rs

```diff
@@ -266,6 +260,29 @@ fn read_until_newline<R: io::BufRead + ?Sized>(
     }
 }
 
+/// Trim contiguous leading and trailing whitespace from `line`.
+///
+/// We consider `\n`, `\r` and ` ` as whitespace.
+fn trim_both_ends(line: &[u8]) -> &[u8] {
+    fn is_whitespace(c: u8) -> bool {
+        matches!(c, b'\n' | b'\r' | b' ')
+    }
+
+    // Walk from the front of the line until the first non-whitespace char.
+    let mut start = 0;
+    while start < line.len() && is_whitespace(line[start]) {
```

Comment:
strip_prefix() is a thing, can we use it here? There's also strip_suffix().

Reply:
I was trying to match the original approach which avoided `strip_suffix`. It looks like you added the previous version in ff11ce1b6b658d498e80515bab6843ed71db938f - was there a reason you didn't use `strip_suffix` there?

This approach has the advantage of not needing two full passes through the string but maybe that's not a very compelling reason for this crate.

Reply:
Not sure I was aware of these then. I was actually looking for something equivalent to `str`'s `trim_matches()`. Nightly now has `slice::trim_ascii()` which I think would be a perfect fit for this?

Probably fine to leave it like this for now.

Reply:
> Probably fine to leave it like this for now.

SGTM

Reply:
Came back to give this a bit more thought.

> strip_prefix() is a thing, can we use it here? There's also strip_suffix().

Looking closer at these stdlib functions I don't think they're a good fit for a couple of reasons. They both take a `SlicePattern<Item=T> + ?Sized` and expect to be given a contiguous pattern of a specific size to remove. We have a couple different needles to look for (`\n`, `\r`, etc) and don't know the length we'll find ahead of time. Using something like `line.strip_suffix(b' ')` doesn't do what you'd hope here and only removes [the first leading space if present](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021).

> I was actually looking for something equivalent to str's trim_matches().

Yeah, that looks better but unfortunately isn't available for `&[u8]`.

> Nightly now has slice::trim_ascii() which I think would be a perfect fit for this?

I could leave a pointer to https://github.com/rust-lang/rust/issues/94035 in a comment for a reminder when it is eventually stabilized and >= MSRV, but I think one potential difference to consider is that this new function uses [u8::is_ascii_whitespace](https://doc.rust-lang.org/std/primitive.u8.html#method.is_ascii_whitespace) which has a broader definition that we've used historically (e.g. including `\t`). That's probably fine, but feels worth calling out.

Reply:
> Call this function trim_ascii(), matching the unstable std method (and linking to the tracking issue)

Done :ballot_box_with_check: 

> Leaning on u8::is_ascii_whitespace(), including \t

Done :ballot_box_with_check: 

> Perhaps copy the matching approach they use?

This was an interesting approach and very unlike how I would sit down to write this, but it seems efficient and if it's good enough for stdlib it's good enough for me :+1: I copied it verbatim with the only modification being folding the two trims into one fn.